testutil/common.go

```diff
@@ -0,0 +1,526 @@
+	params.AppModuleBasic{}.RegisterLegacyAminoCodec(cdc)
+	types.RegisterCodec(cdc)
+	return cdc
+}
+
+// getSubspace returns a param subspace for a given module name.
+func getSubspace(k paramskeeper.Keeper, moduleName string) paramstypes.Subspace {
+	subspace, _ := k.GetSubspace(moduleName)
+	return subspace
+}
+
+// MakeTestMarshaler creates a proto codec for use in testing
+func MakeTestMarshaler() codec.Codec {
+	interfaceRegistry := codectypes.NewInterfaceRegistry()
+	std.RegisterInterfaces(interfaceRegistry)
+	ModuleBasics.RegisterInterfaces(interfaceRegistry)
+	types.RegisterInterfaces(interfaceRegistry)
+	return codec.NewProtoCodec(interfaceRegistry)
+}
+
+// SetupFiveValChain does all the initialization for a 5 Validator chain using the keys here
+func SetupFiveValChain(t *testing.T) (TestInput, sdk.Context) {
+	t.Helper()
+	input := CreateTestEnv(t)
+
+	// Set the params for our modules
+	input.StakingKeeper.SetParams(input.Context, TestingStakeParams)
+
+	// Initialize each of the validators
+	msgServer := stakingkeeper.NewMsgServerImpl(input.StakingKeeper)
+	for i := range []int{0, 1, 2, 3, 4} {
+
+		// Initialize the account for the key
+		acc := input.AccountKeeper.NewAccount(
+			input.Context,
+			authtypes.NewBaseAccount(AccAddrs[i], AccPubKeys[i], uint64(i), 0),
+		)
+
+		// Set the balance for the account
+		require.NoError(t, input.BankKeeper.MintCoins(input.Context, types.ModuleName, InitCoins))
+		// nolint
+		input.BankKeeper.SendCoinsFromModuleToAccount(input.Context, types.ModuleName, acc.GetAddress(), InitCoins)
+
+		// Set the account in state
+		input.AccountKeeper.SetAccount(input.Context, acc)
+
+		// Create a validator for that account using some tokens in the account
+		// and the staking handler
+		_, err := msgServer.CreateValidator(input.Context, NewTestMsgCreateValidator(ValAddrs[i], ConsPubKeys[i], StakingAmount, OrchAddrs[i], EthAddrs[i]))
+
+		// Return error if one exists
+		require.NoError(t, err)
+	}
+
+	// Run the staking endblocker to ensure valset is correct in state
+	staking.EndBlocker(input.Context, input.StakingKeeper)
+
+	// Return the test input
+	return input, input.Context
+}
+
+func NewTestMsgCreateValidator(
+	address sdk.ValAddress,
+	pubKey ccrypto.PubKey,
+	amt cosmosmath.Int,
+	orchAddr sdk.AccAddress,
+	ethAddr gethcommon.Address,
+) *stakingtypes.MsgCreateValidator {
+	commission := stakingtypes.NewCommissionRates(sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec())
+	out, err := stakingtypes.NewMsgCreateValidator(
+		address, pubKey, sdk.NewCoin("stake", amt),
+		stakingtypes.Description{
+			Moniker:         "",
+			Identity:        "",
+			Website:         "",
+			SecurityContact: "",
+			Details:         "",
+		}, commission, sdk.OneInt(),
+		orchAddr, ethAddr,
+	)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
+
+// SetupTestChain sets up a test environment with the provided validator voting weights
+func SetupTestChain(t *testing.T, weights []uint64) (TestInput, sdk.Context) {
```

Comment:
[blocking][new issue needed]

we should ideally consolidate this with https://github.com/celestiaorg/celestia-app/blob/e18de882c3de4ee92e4632af21d4007c1b642adb/testutil/test_app.go#L71

Reply:
Should be done before or after the merge?
I guess it can be done after the merge.

Reply:
yeah, after, but we should create an issue for it so we don't forget

Reply:
https://github.com/celestiaorg/celestia-app/issues/745